.changeset/no-paraemter-assign-unparenthesized-arrow-function.md

```diff
@@ -0,0 +1,15 @@
+---
+"@biomejs/biome": patch
+---
+
+[noParameterAssign](https://biomejs.dev/linter/rules/no-parameter-assign) now reports reassigned parameter of unparenthesized arrow functions.
+
+The following code is now reported as invalid.
+
+```js
+const f = param => {
+  param = {}; // Reassigning a function parameter is confusing.
+};
+```
+
+Fixed [#5409](https://github.com/biomejs/biome/issues/5409).
```

Comment:
for consistency:

```suggestion
Fixed [#5409](https://github.com/biomejs/biome/issues/5409): [noParameterAssign](https://biomejs.dev/linter/rules/no-parameter-assign) now reports reassigned parameter of unparenthesized arrow functions.

The following code is now reported as invalid.

```js
const f = param => {
  param = {}; // Reassigning a function parameter is confusing.
};
```